File: lista python/at12.py
class Produto:
    def __init__(self, nome, preco):
        self.nome = nome
        self.preco = preco

    
    def __add__(self, outro_produto):
        return self.preco + outro_produto.preco

    def __str__(self):
        return f"{self.nome} - R${self.preco:.2f}"


produto1 = Produto("Produto X", 90.75)
produto2 = Produto("Produto Y", 300.25)


soma_precos = produto1 + produto2

print(f"Soma dos preços: R${soma_precos:.2f}")

File: lista java/at12.java
public class Produto {
    private String nome;
    private double preco;

    public Produto(String nome, double preco) {
        this.nome = nome;
        this.preco = preco;
    }

 
    public double somarPreco(Produto outroProduto) {
        return this.preco + outroProduto.preco;
    }

    @Override
    public String toString() {
        return nome + " - R$" + String.format("%.2f", preco);
    }

    public static void main(String[] args) {
        Produto produto1 = new Produto("Produto A", 50.75);
        Produto produto2 = new Produto("Produto B", 100.25);

     
        double somaPrecos = produto1.somarPreco(produto2);

        System.out.println("Soma dos preços: R$" + String.format("%.2f", somaPrecos));
    }
}
